File: javascript/grade-school/grade-school.js
var School = function(){
  this.schoolRoster = {};
}

School.prototype.roster = function(){
  return this.schoolRoster;
};

School.prototype.add = function(student, grade){
  this.schoolRoster[grade] = this.schoolRoster[grade] || [];
  this.schoolRoster[grade].push(student);
  this.schoolRoster[grade].sort();
}

School.prototype.grade = function(gradeNumber){
  return this.schoolRoster[gradeNumber] || [];
}

module.exports = School;

File: python/grade-school/grade_school.py
from collections import defaultdict

class School():
    def __init__(self, school_name):

        self.school_name = school_name
        self._grade_levels = defaultdict(list)


    def sort(self):
        return ((key, tuple(self.grade(key))) for key in sorted(self._grade_levels.keys()))


    def grade(self, grade_level):
        return sorted(self._grade_levels[grade_level])


    def add(self, student, grade_level):
        self._grade_levels[grade_level].append(student)
